### PR TITLE
[gha] Update release action using grimoirelab-github-actions

### PR DESCRIPTION
**Pull request by jjmerchante**
This PR updates the release.yml file to include the new [grimoirelab-github-actions](https://github.com/chaoss/grimoirelab-github-actions)

Related to https://github.com/chaoss/grimoirelab/issues/463

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -11,24 +11,11 @@ jobs:
...
-          poetry build
-      - name: Upload distribution artifacts
-        uses: actions/upload-artifact@v2
+      - name: "Build package using Poetry and store result."
+        uses: chaoss/grimoirelab-github-actions/build@master
         with:
-          name: toolkit-dist
-          path: dist
-  
+          artifact-name: grimoire-dist
```
Comment (vchrombie) [inline .github/workflows/release.yml:17]: How about changing the `artifact-name`
```
          artifact-name: toolkit-dist
```

If we want to have the same name across the grimoirelab repos, then I think we need to also update the [artifact name in the `tests` step](https://github.com/chaoss/grimoirelab-toolkit/blob/e5a8ad1d94da569f8c94c92b98cd027103c882aa/.github/workflows/release.yml#L43).

Comment (jjmerchante) [reply .github/workflows/release.yml]: Yes, toolkit-dist its a better name :ok_hand:

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -11,24 +11,12 @@ jobs:
...
-      - name: Install poetry
-        run: |
-          curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python -
-          echo "PATH=$HOME/.poetry/bin:$PATH" >> $GITHUB_ENV
-      - name: Build distributions
-        run: |
-          poetry build
-      - name: Upload distribution artifacts
-        uses: actions/upload-artifact@v2
+      - name: "Build package using Poetry and store result."
```
Comment (vchrombie) [inline .github/workflows/release.yml:14]: @jjmerchante I know this review comment might be silly :speak_no_evil: 
Just to maintain the uniformity, do you mind removing the quotes and full stop. I mean we haven't added it anywhere in the workflows till now. The name key takes the value as a string, so we need not specify it.

I'm sorry for being so picky :sweat_smile::pray:

Comment (jjmerchante) [reply .github/workflows/release.yml]: Yes, I didn't notice it! :+1:

**Comment by vchrombie** [comment]
@sduenas do you wanna add a review too?

File: .github/workflows/release.yml (hunk 2 above)
Comment (vchrombie) [reply .github/workflows/release.yml]: Thanks JJ!! :partying_face:

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -51,48 +39,23 @@ jobs:
...
     steps:
-      - uses: actions/checkout@v2
-      - name: Download distribution artifact
-        uses: actions/download-artifact@v2
-        with:
-          name: toolkit-dist
-          path: dist
-      - name: Set up Python
-        uses: actions/setup-python@v2
+      - name: Publish the package in PyPI
```
Comment (sduenas) [inline .github/workflows/release.yml:56]:
```suggestion
      - name: Publish the package on PyPI
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -51,48 +39,23 @@ jobs:
...
-      - uses: actions/checkout@v2
-      - name: Get release tag
-        id: tag
-        run: |
-          echo ::set-output name=tag::${GITHUB_REF#refs/tags/}
-      - name: Create release
-        uses: actions/create-release@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Create a new release for the repository
```
Comment (sduenas) [inline .github/workflows/release.yml:47]:
```suggestion
      - name: Create a new release on the repository
```

**Review by sduenas — CHANGES_REQUESTED** [review]
LGTM but just correct the English in some of the titles.

**Review by sduenas — APPROVED** [review]
LGTM

**Pull request closed without merging**